src/coreclr/vm/interoplibinterface_objc.cpp

```diff
@@ -11,20 +11,68 @@
 
 #include "interoplibinterface.h"
 
-BOOL QCALLTYPE ObjCWrappersNative::TrySetGlobalMessageSendCallbacks(
-    _In_ void* fptr_objc_msgSend,
-    _In_ void* fptr_objc_msgSend_fpret,
-    _In_ void* fptr_objc_msgSend_stret,
-    _In_ void* fptr_objc_msgSendSuper,
-    _In_ void* fptr_objc_msgSendSuper_stret)
+#include <pinvokeoverride.h>
+
+#define OBJC_MSGSEND "objc_msgSend"
+
+namespace
+{
+    BOOL s_msgSendOverridden = FALSE;
+    void* s_msgSendOverrides[ObjCWrappersNative::MsgSendFunction::Last + 1] = {0};
+
+    const char* ObjectiveCLibrary = "/usr/lib/libobjc.dylib";
```

Comment:
Took this from:
https://github.com/xamarin/xamarin-macios/blob/13fd8f5b07361a14186d77e2b1719449414edb2f/runtime/runtime.m#L2451-L2454

The [doc](https://docs.microsoft.com/xamarin/ios/internals/objective-c-selectors) examples use `Constants.ObjectiveCLibrary`:
https://github.com/xamarin/xamarin-macios/blob/13fd8f5b07361a14186d77e2b1719449414edb2f/src/ObjCRuntime/Constants.cs#L9

Do we care to do all the possible variations too? (e.g. `objc`, `objc.dylib`, `libobjc`, `libobjc.dylib`)

Reply:
I am fine with the strict supported scenario for this. It is a quirk behavior so we can be strict.